core/server/controllers/frontend/index.js

```diff
@@ -67,7 +67,10 @@ frontendControllers = {
             }
 
             // CASE: we only support /:slug format for pages
-            if (post.page && post.url !== req.path) {
+            // However, given that a url with `/edit/` appended is different
+            // than the post url, we must remove `/edit/` from the url if it
+            // is an edit url.
+            if (post.page && post.url !== ((lookup.isEditURL) ? req.path.replace(/edit\/$/, '') : req.path)) {
```

Comment:
Honestly I would agree, but I don't think it's super simple to do that.

In the [post lookup method](https://github.com/TryGhost/Ghost/blob/master/core/server/controllers/frontend/post-lookup.js#L26) - we compare the url from the request to the `config.theme.permalinks` value - and there are two things that can happen:

1. Date permalinks are turned off - and a user navigates to the page url. The postLookup function would match the post permalink pattern, because it's identical to the page lookup pattern
2. Date permalinks are turned on - and a user navigates to the page url with a date prepended. This would still match the post lookup pattern, leaving no way to determine if the user is looking up a post or a page.

Now, we can do the comparison after the `api.posts.read` call - which knows whether or not the post is a page or not; however, in order to determine fully if the user is accessing it via the correct path, we'd have to know somehow whether the value of `config.theme.permalinks` had the date in it, leading to more complexity, especially if we end up supporting custom permalinks at some point.